README.md

```diff
@@ -260,6 +260,46 @@ Additionally, `GraphQlQueryResponseData` has been exposed to users:
 import type { GraphQlQueryResponseData } from "@octokit/graphql";
 ```
 
+### Usage with graphql-codegen
+
+If your query is represented as a `String & DocumentTypeDecoration`, for example as [`TypedDocumentString` from graphql-codegen and its client preset](https://the-guild.dev/graphql/codegen/docs/guides/vanilla-typescript), then you can get type-safety for the query's parameters and return values.
+
+Assuming you have configured graphql-codegen, with [GitHub's GraphQL schema](https://docs.github.com/en/graphql/overview/public-schema), and an output under `./graphql`:
```

Comment:
I'm trying to balance giving a complete guide to graphql-codegen and the GitHub GraphQL API, but maybe we can be more descriptive here 💭